### ISSUE TITLE
npm project incompatible with Windows

### ISSUE THREAD

**Issue by zacharysang** [RevolutionUC/revolutionuc-website#11]
the scripts entry in the package.json includes code exclusive to unix systems. Commands such as 'npm start' or 'npm run dev' do not currently work on Windows

**Comment by domfarolino** [comment]
What errors do you get in the console on windows

**Comment by zacharysang** [comment]
First error we hit is: concurrently' is not recognized as an internal or external command, but there are a few other keywords in there that will throw the same error

**Comment by domfarolino** [comment]
You did `npm install`? `concurrently` is just an npm module, is why I ask

**Comment by zacharysang** [comment]
Yeah, my mistake. I forgot to run npm install after cloning again. But there are still some issues with Windows compatibility showing up

**Comment by domfarolino** [comment]
Are you straight up running this in CMD, or the Git bash?

**Comment by isiah-lloyd** [comment]
This may be it's own issue but I'm running this using Bash on Ubuntu on Windows. And I get this error everytime a file is updated: https://gist.github.com/isiah-lloyd/a7d0cda28f845d5b7551f27d9d4852e0
I have to run `npm run dev` again to start the server. I have updated NPM and Node

**Comment by domfarolino** [comment]
"on Ubuntu on Windows"...VM? @isiah-lloyd

**Comment by isiah-lloyd** [comment]
@domfarolino Sorry, that's literally what they decided to call the [Bash implementation on Windows](https://msdn.microsoft.com/en-us/commandline/wsl/about)

**Comment by domfarolino** [comment]
Ohhh so you're running git bash.

@zacharysang @isiah-lloyd :

There seems to be an issue with `concurrently` on windows..or something idk. We're using concurrently to run two commands at once basically but better yet we're using the `--kill-others` flag to kill one task if the other dies. Maybe try the following:

 - Edit the package json to remove the `--kill-others` flag and try and run again
 - If the above fails (probably will) try running the two npm commands separately in different windows
 - If the above works, then `concurrently` is the problem.

Basically I want to narrow it down to see if the webpack/server command is failing, or if `conurrently` is failing on its own. I'm hoping the two commands independently work and `concurrently` has issues. Else we've got slightly larger problems.

**Comment by isiah-lloyd** [comment]
Option one worked for me. No longer crashes.

**Comment by domfarolino** [comment]
Damn. That probably means when you end one of the processes, the other will still technically fail. Not sure though. I guess `npm run dev` twice in a row (after stopping it of course) will let you know if it fails trying to start up a process that might already exist. Also if running it twice in a row doesn't fail, I'm wondering if it is accumulating processes on your computer. That would suck

**Comment by isiah-lloyd** [comment]
I'm no expert on Linux but when I run `top` I see no processes from node running after running `npm run dev` many times

**Comment by domfarolino** [comment]
I usually use
`netstat -tulpn` for node processes
and check `ps -aef | grep 'webpack'` and see if anything with `webpack` and `--watch` are running but really if you're not having issues or seeing the processes you're probably a-ok. ;)

**Comment by domfarolino** [comment]
I can close this correct? Isiah you got this working on windows, which means zach should be able to as well?

**Comment by isiah-lloyd** [comment]
Yes

On Jan 15, 2017 4:24 PM, "Dominic Farolino" <notifications@github.com>
wrote:

I can close this correct? Isiah you got this working on windows, which
means zach should be able to as well?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/RevolutionUC/revolutionuc-website/issues/11#issuecomment-272724924>,
or mute the thread
